src/segmentation.rs

```diff
@@ -0,0 +1,102 @@
+//! Segmenter trait and default implementations.
```

Comment:
Let's say I'm parsing simple ASCII text: what's the recommended default segmentation strategy I should use after ASCII filtering and (perhaps) another normalization strategy is applied?

Reply:
I'm not sure I would recommend ASCII filtering. It would be dependent on the language and typical unicode behaviour.

I could imagine three 'defaults':

1. CJK/ideographic languages - CJK + Form C and Grapheme
2. Simpler Latin-ish languages (horribly non-technical description) - UnicodeToAsciiNormalization maybe? and Codepoint (well, if its ascii it can probably just be bytes... i dont remember if extended ascii is still just a byte and whether `is_ascii` accepts those...)
3. Other (e.g., Cyrillic languages, etc) - probably Form C + Grapheme

these need more research and are based off of very, very broad ideas about language structures and use. I'm assuming that if someone is processing spanish, they are okay with mañana = manana if they're doing fuzzy matching. That might not be true!

asian languages might be better off with codepoint segmentation after normalization (i doubt it, but idk)

There probably needs to be some documentation on this but a lot of it will merely be to point out the trade offs of the various normalization and segmentation strategies so that users can pick. Though we can probably make some broad recommendations like these for less sophisticated users.

Reply:
> Let's say I'm parsing simple ASCII text

I completely glazed over this qualifier.

Reply:
The motivation behind my question is the decision for default behavior for the current public interface of the library. I'm assuming that if a user wants to customize their normalization and segmentation strategy through this new code, there will be additional public functions added that allow for this customization... meaning we'll need to select a default behavior (likely whatever we have now?) for the existing API.

> these need more research and are based off of very, very broad ideas about language structures and use.

Agreed, and I'm not sure where to begin for making recommendations to library users about what sort of seg/norm they need.

Reply:
That's valid. I want to defer that conversation for now. These implementations can be stable without making that choice. I'm expecting I might make a `fuzzywuzzy_compat` module that will be our normal functions with fuzzywuzzy compatible defaults - whatever those may be, but I suspect it'll be code point segmentation.

Reply:
To explicitly call it out, I'm okay with making a breaking change as this is pre-1.0 and we are on a *relatively* short path to 1.0.